core/client/src/main/java/alluxio/client/block/stream/NettyPacketWriter.java

```diff
@@ -49,10 +49,13 @@
  *    buffer is full, resumes if the buffer is not full.
  * 2. The server reads packets from the channel and writes them to the block worker. See the server
  *    side implementation for details.
- * 3. When all the packets are sent, the client closes the reader by sending an empty packet to
- *    the server to signify the end of the block. The client must wait the response from the server
- *    to make sure everything has been written to the block worker.
+ * 3. The client can either send an EOF packet or a CANCEL packet to end the write request. The
+ *    client has to wait for the response from the data server for the EOF or CANCEL packet to make
+ *    sure that the server has cleaned its states.
  * 4. To make it simple to handle errors, the channel is closed if any error occurs.
+ *
+ * NOTE: this class is NOT threadsafe. Do not call cancel/close while some other threads are
+ * writing.
```

Comment:
I think it is just same as any other non-threadsafe classes. The user can use it in multi thread as long as they have proper synchronization. I am emphasizing its non-threadsafty here.